File: Core/DTOs/BorrowalReturnInfoDto.cs
namespace Core.DTOs
{
    public class BorrowalReturnInfoDto : ResultDto
    {
        public int BorrowalId { get; set; }
        public BorrowalsDisplayDto? BorrowalsDisplayDto { get; set; }

        public string BorrowalDateStr { get; set; } = string.Empty;
        public string DueDateStr { get; set; } = string.Empty;

        public bool IsOverdue { get; set; }
        public int LateDays { get; set; }
        public decimal PerDayLateFeeDollars { get; set; }        
        public decimal AmountDue => LateDays * PerDayLateFeeDollars;
    }
}

File: client/src/app/shared/models/borrowal-return-info-dto.ts
import { BorrowalsDisplayDto } from "./borrowals-display-dto";
import { ResultDto } from "./result-dto";

export interface BorrowalReturnInfoDto extends ResultDto {
    borrowalId: number;
    borrowalsDisplayDto?: BorrowalsDisplayDto; 
  
    borrowalDateStr: string;
    dueDateStr: string;
  
    isOverdue: boolean;
    lateDays: number;
    perDayLateFeeDollars: number;
  
    amountDue: number; 
  }